src/hooks/useFarmStats.ts

```diff
@@ -27,87 +27,78 @@ function buildFarmStatsQuery(farmAddress: string) {
+  const [totalStaked, setTotalStaked] = useState(ZERO_BIG_INT);
+  const [rewardUnlockRate, setRewardUnlockRate] = useState(ZERO_BIG_INT);
+  const [timeRemaining, setTimeRemaining] = useState({});
 
   const FARM_STATS_QUERY = buildFarmStatsQuery(farmAddress);
 
   useEffect(() => {
-    let cancelled = false
-    let retryTimer: NodeJS.Timeout
-    let farmDaysLeft = 0
+    let cancelled = false;
+    let retryTimer: NodeJS.Timeout;
 
     async function fetchFarmStats() {
       try {
-        const result = await graphqlClient.query(FARM_STATS_QUERY).toPromise()
+        const result = await graphqlClient.query(FARM_STATS_QUERY).toPromise();
 
         if (!result?.data) {
-          return
+          return;
         }
 
-        const [{ 
-          apy, 
-          tvl,
-          staked, 
-          bonusPeriodSec,
-          createdTimestamp,
-          totalUnlockedRewards, 
-        }]: [FarmStatsResponse] = result.data.tokenGeysers
-        
+        const [{ apy, tvl, staked, bonusPeriodSec, createdTimestamp, totalUnlockedRewards }]: [
+          FarmStatsResponse,
+        ] = result.data.tokenGeysers;
+
         const parsedAPY = BigInt(apy);
         const parsedTVL = BigInt(tvl);
         const parsedStaked = BigInt(staked);
 
         const parsedTotalUnlockedRewards = BigInt(totalUnlockedRewards);
 
         const unlockRate = parsedTotalUnlockedRewards / BigInt(DAYS_IN_ONE_MONTH);
-        
-        const today = new Date();
-        
+
+        const today = dayjs();
+
         // Calculate farm end date based on the bonus period (at this time, all rewards are unlocked)
-        const farmEndDate = new Date((createdTimestamp + bonusPeriodSec) * 1000)
-        const farmTimeLeft = farmEndDate.getTime() - today.getTime();
+        const farmEndDate = dayjs.unix(createdTimestamp + bonusPeriodSec);
+        const farmTimeLeft = farmEndDate.diff(today, 'seconds');
 
-        // Return days left, if any (otherwise, remains at 0)
-        if (farmTimeLeft > 0) {
-          farmDaysLeft = Math.ceil(farmTimeLeft / ONE_DAY); 
-        }
+        const timeRemaining: TimeRemaining = getTimeRemaining(farmTimeLeft);
 
         if (!cancelled) {
-          setAPY(parsedAPY)
-          setTVL(parsedTVL)
-          setTotalStaked(parsedStaked)
-          setDaysLeft(farmDaysLeft)
-          setRewardUnlockRate(unlockRate)
+          setAPY(parsedAPY);
+          setTVL(parsedTVL);
+          setTotalStaked(parsedStaked);
+          setTimeRemaining(timeRemaining);
+          setRewardUnlockRate(unlockRate);
         }
       } catch (err) {
-        retryTimer = setTimeout(fetchFarmStats, RETRY_EVERY)
+        retryTimer = setTimeout(fetchFarmStats, RETRY_EVERY);
       }
     }
 
-    fetchFarmStats()
+    fetchFarmStats();
 
     return () => {
-      cancelled = true
-      clearTimeout(retryTimer)
-    }
-  }, [FARM_STATS_QUERY])
+      cancelled = true;
+      clearTimeout(retryTimer);
+    };
+  }, [FARM_STATS_QUERY]);
```

Comment:
This is more of a question, but what does the `gql` tag for `FARM_STATS_QUERY` resolve to? If it's an object, we might have an issue here for re-rendering and might need to `.stringify()` it.